index.html

```diff
@@ -917,40 +848,31 @@ <h2>
         checkbox.setAttribute("type", "checkbox");
         document.body.appendChild(checkbox);
 
-        const lock = new WakeLock("screen");
-
-        lock.onactivechange = () =&gt; checkbox.checked = lock.active;
+        let controller;
 
-        const controller = new AbortController();
-
-        try {
-          checkbox.disabled = true;
-          if (checkbox.checked) {
-            await lock.request({ signal: controller.signal });
-          } else {
-            controller.abort();
+        checkbox.onclick = () =&gt; {
+          try {
+            if (checkbox.checked) {
+              controller = new AbortController();
+              WakeLock.request("screen", { signal: controller.signal });
```

Comment:
`await` is needed here so that the Promise rejection triggers the `catch` block.

Reply:
Fixed in #210.